src/states/lessonIndexState.ts

```diff
@@ -0,0 +1,44 @@
+import { atomWithLazy, unwrap } from "jotai/utils";
+import { useAtomValue } from "jotai";
+import { LessonIndexEntry } from "../types";
+
+const lessonIndexFallback: LessonIndexEntry[] = [{
+  "title": "Steno",
+  "subtitle": "",
+  "category": "Drills",
+  "subcategory": "",
+  "path": process.env.PUBLIC_URL + "/drills/steno/lesson.txt",
+  overview: "",
+  suggestedNext: "",
+  wordCount: 0,
+}];
+
+function fetchLessonIndex() {
+  return fetch(process.env.PUBLIC_URL + '/lessons/lessonIndex.json', {
+    method: "GET",
+    credentials: "same-origin"
+  }).then((response) => {
+    return response.json()
+  }).then(json => {
+    return(json);
+  }).catch(() => {
+    return lessonIndexFallback;
+  });
+}
+
+const _lessonIndexState = atomWithLazy<Promise<LessonIndexEntry[]>>(fetchLessonIndex);
+
+/**
+ * This suspends while loading
+ */
+export const useLessonIndex = () => {
+  return useAtomValue(_lessonIndexState);
+}
+
+/**
+ * This always returns some value. fallback while loading
+ */
+export const useLessonIndexWithFallback = () => {
```

Comment:
There are two versions for lesson index:

* `useLessonIndexWithFallback` keeps the existing fallback logic before the lesson index has loaded, and never suspends. I kept the usage of this for `App` where it won't be necessarily used right away.
* `useLessonIndex` above suspends before loading is finished. In case lesson index is the main content, using this is better because showing fallback would be confusing for a short period of time